app/src/main/java/fr/free/nrw/commons/category/CategorizationFragment.java

```diff
@@ -320,8 +321,8 @@ protected void onPostExecute(ArrayList<String> result) {
                 latch.countDown();
             }
         };
-        prefixUpdaterSub.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
-        methodAUpdaterSub.executeOnExecutor(AsyncTask.THREAD_POOL_EXECUTOR);
+        prefixUpdaterSub.executeOnExecutor(Executors.newSingleThreadExecutor());
+        methodAUpdaterSub.executeOnExecutor(Executors.newSingleThreadExecutor());
     }
```

Comment:
It appears that this change will mean that we are creating as many executors as characters you type. If so, isn't that spending too much resource than needed?